x-pack/metricbeat/module/openai/usage/persistcache.go

```diff
@@ -0,0 +1,172 @@
+	if strings.TrimSpace(storePath) == "" {
+		return nil, errors.New("empty path provided")
+	}
+
+	store, err := newStateStore(storePath)
+	if err != nil {
+		return nil, fmt.Errorf("create state store: %w", err)
+	}
+
+	return &stateManager{
+		store:     store,
+		keyPrefix: "state_",
+		hashCache: sync.Map{},
+	}, nil
+}
+
+// newStateStore creates a new state store instance at the specified path
+func newStateStore(path string) (*stateStore, error) {
+	if err := os.MkdirAll(path, 0o755); err != nil {
+		return nil, fmt.Errorf("creating state directory: %w", err)
+	}
+	return &stateStore{
+		Dir: path,
+	}, nil
+}
+
+// getStatePath builds the full file path for a given state key
+func (s *stateStore) getStatePath(name string) string {
+	return path.Join(s.Dir, name)
+}
+
+// Put stores a value in a file named by the key
+func (s *stateStore) Put(key string, value string) error {
+	s.Lock()
+	defer s.Unlock()
+
+	filePath := s.getStatePath(key)
+
+	// In case the file already exists, file is truncated.
+	f, err := os.Create(filePath)
+	if err != nil {
+		return fmt.Errorf("creating state file: %w", err)
+	}
+	defer f.Close()
+
+	_, err = f.WriteString(value)
+	if err != nil {
+		return fmt.Errorf("writing value to state file: %w", err)
+	}
+
+	if err = f.Sync(); err != nil {
+		return fmt.Errorf("syncing state file: %w", err)
+	}
+
+	return nil
+}
+
+// Get retrieves the value stored in the file named by the key
+func (s *stateStore) Get(key string) (string, error) {
+	s.RLock()
+	defer s.RUnlock()
+
+	filePath := s.getStatePath(key)
+	data, err := os.ReadFile(filePath)
+	if err != nil {
+		return "", fmt.Errorf("reading state file: %w", err)
+	}
+	return string(data), nil
+}
+
+// Has checks if a state exists for the given key
+func (s *stateStore) Has(key string) bool {
+	s.RLock()
+	defer s.RUnlock()
+
+	filePath := s.getStatePath(key)
+	_, err := os.Stat(filePath)
+	return err == nil
+}
+
+// Remove deletes the state file for the given key
+func (s *stateStore) Remove(key string) error {
+	s.Lock()
+	defer s.Unlock()
+
+	filePath := s.getStatePath(key)
+	if err := os.Remove(filePath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("removing state file: %w", err)
+	}
+	return nil
+}
+
+// Clear removes all state files by deleting and recreating the state directory
+func (s *stateStore) Clear() error {
+	s.Lock()
+	defer s.Unlock()
+
+	if err := os.RemoveAll(s.Dir); err != nil {
+		return fmt.Errorf("clearing state directory: %w", err)
+	}
+	return os.MkdirAll(s.Dir, 0o755)
+}
+
+// GetLastProcessedDate retrieves and parses the last processed date for a given API key
+func (s *stateManager) GetLastProcessedDate(apiKey string) (time.Time, error) {
+	stateKey := s.GetStateKey(apiKey)
+
+	if !s.store.Has(stateKey) {
+		return time.Time{}, ErrNoState
+	}
+
+	dateStr, err := s.store.Get(stateKey)
+	if err != nil {
+		return time.Time{}, fmt.Errorf("get state: %w", err)
+	}
+
+	return time.Parse("2006-01-02", dateStr)
```

Comment:
For better maintainability
 
const dateFormat = "2006-01-02"
return time.Parse(dateFormat, dateStr)